src/carnot/planner/objects/pixie_module.h

```diff
@@ -200,6 +200,29 @@ class PixieModule : public QLObject {
   Returns:
     px.Time: The time value represented in the data.
   )doc";
+
+  inline static constexpr char kFormatDurationOpID[] = "format_duration";
+  inline static constexpr char kFormatDurationDocstring[] = R"doc(
+  Convert an integer in nanoseconds (-3,000,000,000) to a duration string ("-5m") while
+  preserving the sign. This function converts to whole milliseconds, seconds, minutes,
+  hours or days. This means it will round down to the nearest whole number time unit.
```

Comment:
I'm not sure I agree with this behaviour. eg. I don't think `px.format_duration(int representing 5h59m) == '5h'` makes much sense

Reply:
I should have explained this in the summary of the PR, but I chose to implement it this way because our UI only seems to supports a duration string with a single time unit.

You can see from the screenshots below that specifying `-1d12h` and `-36h` yield different results. The former seems to return results from the last 24 hours, which seems to indicate that the `12h` portion is ignored.

<img width="1728" alt="Screen Shot 2023-05-01 at 5 23 00 PM" src="https://user-images.githubusercontent.com/5855593/235554064-5ed3b77a-bce6-4e80-bad7-90289816ad61.png">
<img width="1728" alt="Screen Shot 2023-05-01 at 5 23 13 PM" src="https://user-images.githubusercontent.com/5855593/235554068-bc25f01f-7cea-48fa-8562-432c217fb365.png">

In addition to that, the parse_duration function will only parse strings that include a [single](https://github.com/pixie-io/pixie/blob/0072ce95cde9d6a8788163746a41e833482136be/src/carnot/planner/objects/pixie_module.cc#L371) time [unit](https://github.com/pixie-io/pixie/blob/0072ce95cde9d6a8788163746a41e833482136be/src/common/base/time.h#L42).

I essentially was aiming for compatibility with this change. Since the function's use case would be to craft other pxl script time windows, I wanted it to support the inputs our pxl scripts and related functions (`parse_duration`) expect.

I'm happy to change this but it feels like there is a potential to cause confusion. If we want to pursue that, I would want to evaluate how much work it would be to support that type of a duration string in the other aspects of Pixie (UI and other pxl script functions).

Reply:
Makes sense, I didn't realize that `parse_duration` doesn't handle this case either. We can maybe consider fixing this across the board later. But for now I think mirroring what `parse_duration` does is fine.